Ship_Game/Universe/SolarBodies/ExoticSystemsListScreenItem.cs

```diff
@@ -0,0 +1,225 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using Ship_Game.AI;
+using Ship_Game.Audio;
+using Ship_Game.Commands.Goals;
+using Ship_Game.Ships;
+using System.Linq;
+using SDGraphics;
+using SDUtils;
+using Vector2 = SDGraphics.Vector2;
+using Rectangle = SDGraphics.Rectangle;
+using Ship_Game.Graphics;
+using Ship_Game.Universe.SolarBodies;
+using Ship_Game.Universe;
+using System;
+
+namespace Ship_Game
+{
+    public sealed class ExoticSystemsListScreenItem : ScrollListItem<ExoticSystemsListScreenItem> // Moved to UI V2
+    {
+        public readonly Planet Planet;
+        public readonly SolarSystem System;
+        public Rectangle SysNameRect;
+        public Rectangle PlanetNameRect;
+        public Rectangle OrdersRect;
+        public Rectangle DistanceRect;
+
+        Empire Player => Universe.Player;
+        readonly Color Cream = Colors.Cream;
+        readonly Graphics.Font NormalFont = Fonts.Arial20Bold;
+        readonly Graphics.Font SmallFont = Fonts.Arial12Bold;
+        readonly Graphics.Font TinyFont = Fonts.Arial8Bold;
+        readonly Color TextColor = new Color(255, 239, 208);
+
+        Rectangle PlanetIconRect;
+        readonly UITextEntry PlanetNameEntry = new UITextEntry();
+        UIButton Research;
+        readonly float Distance;
+        bool MarkedForResearch;
+        readonly UniverseState Universe;
+
+        UITextEntry ResearchTextInfo;
+        bool IsPlanet => Planet != null;
+        public bool IsStar => Planet == null;
+        ExplorableGameObject SolarBody;
+
+        public ExoticSystemsListScreenItem(ExplorableGameObject solarBody, float distance)
+        {
+            SolarBody = solarBody;
+            if (solarBody is Planet planet) 
+            {
+                Planet   = planet;
+                System   = planet.System;
+                Universe = planet.Universe;
+            }
+            else
+            {
+                SolarSystem system = solarBody as SolarSystem;
+                System   = system;
+                Universe = system.Universe;
+            }
+
+            Distance = distance / 1000; // Distance from nearest player colony
+            foreach (Goal g in Player.AI.Goals)
+            {
+                if (solarBody.IsResearchable 
+                    && !solarBody.IsResearchStationDeployedBy(Player)
+                    && g.IsResearchStationGoal(solarBody))
+                {
+                    MarkedForResearch = true;
+                }
+            }
+        }
+
+        public override void PerformLayout()
+        {
+            int x = (int)X;
+            int y = (int)Y;
+            int w = (int)Width;
+            int h = (int)Height;
+            RemoveAll();
```

Comment:
You could also use relative coordinate systems or something, so if you move one element, all the other elements will move along with it. Is the RemoveAll() strictly necessary?